src/vsr.zig

```diff
@@ -1917,3 +1918,52 @@ const ViewChangeHeadersArray = struct {
         headers.array.appendAssumeCapacity(Headers.dvc_blank(op));
     }
 };
+
+pub const Op = struct {
+    pub fn checkpoint_before(checkpoint: u64) ?u64 {
+        assert(checkpoint_valid(checkpoint));
+
+        if (checkpoint == 0) {
+            return null;
+        }
```

Comment:
```suggestion
        if (checkpoint == 0) return null;
```

If it fits on a single line, it's okay to do a one liner return. I know the kernel doesn't like this, but it's not something we enforce because it can make for leaner code, according to context (for example, in the state machine).